### ISSUE TITLE
UserId

### ISSUE THREAD

**Issue by rmeloni** [nielsgl/sequelize-paper-trail#70]
I added a userModel, and userId opt at update/create (show at debug) but not insert userId at Revisions.

The Revisions table has a userdbid column.

var opts = {
  //underscored: true,
  //underscoredAttributes: true,
  //enableCompression: false,
  //enableMigration: true,
  **userModel: 'userdb'**
  ,debug: true
};

afterHook called
instance: roledb {
  dataValues: 
   { ............ },
  _previousDataValues: 
   { ........ },
  _changed: { structureread: true, structurewrite: true, revision: true },
  _modelOptions: 
   { ........ },
  _options: 
   { ........ },
  __eagerlyLoadedAssociations: [],
  isNewRecord: false,
  context: { delta: [ [Object], [Object] ] } }
opt: { **userId: 2,**
  fields: [ 'structureread', 'structurewrite', 'updatedAt', 'revision' ],
  defaultFields: [ 'structureread', 'structurewrite', 'updatedAt' ],
  hooks: true,
  validate: true }

**Comment by lmorningstar116** [comment]
Faced the same problem.

Pass the userModel to options. add userId opt at update.

But the query is executed without userId.

`INSERT INTO "Revisions" ("id","model","document","operation","documentId","revision","createdAt","updatedAt") VALUES (DEFAULT,'OrganizationSettings','{"id":1,"use_fisca
l_printer":true,"fiscal_printer_service_type":"atol","use_scanner":false,"scanner_service_uri":"","revision":10,"organization_id":1}','update',1,10,'2019-03-02 12:14:06.283 +00:00','2019-03
-02 12:14:06.283 +00:00') RETURNING *;`

**Comment by nielsgl** [comment]
@rmeloni @lmorningstar116 are you using the continuation local storage package?

**Comment by rmeloni** [comment]
No...

The problem is that the Revision.build() use "userId" field and my table name is userdb and create a userdbid field at Revision table.

I changed userId: ns && ns.get(options.continuationKey) || opt['userId'], to userdbId: ns && ns.get(options.continuationKey) || opt['userId'], at query and works.

**Comment by nielsgl** [comment]
Ah I get it, I think if I add a `userIdAttribute` that you can set to `userdbId` in your case and then I will use that in that line it should pick the right attribute 🤔

**Comment by nielsgl** [comment]
@rmeloni can you try it by installing the package from this github commit:
```
git://github.com/nielsgl/sequelize-paper-trail.git#e62c103bac72cb848018ef1554bd5dbb64d8ee20
```

**Comment by rmeloni** [comment]
@nielsgl this github has a change for previous and current version.  Is it github?

**Comment by nielsgl** [comment]
I'm not sure what you mean. You can find the code here: https://github.com/nielsgl/sequelize-paper-trail/blob/feature/refactoring/lib/index.js

**Comment by rmeloni** [comment]
It´s works!

**Comment by nielsgl** [comment]
Awesome, I'm finishing up that branch and will release it tomorrow as a next major release (v3). I'll drop a comment here when it's done so you can upgrade it and install it as a normal npm package.

**Comment by nielsgl** [comment]
@rmeloni it's available in `sequelize-paper-trail@3.0.0-rc.3`
